modules/data.remote/R/merge_image_tiles.R

```diff
@@ -102,4 +102,75 @@ merge_image_tiles <- function(folder.path,
     unlink(list.files(folder.path, full.names = T)[which(!grepl("merged_image.tif", list.files(folder.path)))], recursive = T)
   }
   return(file.path(folder.path, paste0("merged_image.tif")))
+}
+
+#' @description This function provides tool for remote sensing image conversion using GDAL utility.
+#' @details
+#' Please note that, this function only supports conversions for one band of one image.
+#' If you want to convert multiple images or bands, make sure to loop over these targets.
+#' Currently tested H5, NetCDF, HDF4, and GeoTIFF formats. 
+#' This function should be ready to any GDAL supported image format.
+#' 
+#' @title gdal_conversion
+#' @param in_path character: physical path to the image file.
+#' @param outfolder character: physical path to the folder where you want to export the converted image. Default is NULL.
+#' @param band_name character: band name of the image. Default is NULL.
+#' @param just_band_name boolean: if we just want the band names of the image file. Default is TRUE.
+#' @param target_format character: target image format. Default is .tif.
+#' @author Dongchen Zhang.
+#' @examples
+#' \dontrun{
+#' in_path <- "/projectnb/dietzelab/malmborg/CARB/HLS_data/MSLSP_10SDH_2016.nc"
+#' outfolder <- "/projectnb/dietzelab/dongchen/anchorSites/NA_runs/MODIS_Phenology"
+#' band_name <- "NumCycles"
+#' # try grab all available bands from the target file.
+#' band_names <- 
+#'   gdal_conversion(in_path = in_path, 
+#'     outfolder = outfolder, 
+#'     band_name = NULL, 
+#'     just_band_name = T)
+#' # try convert the first band of the available band names to GeoTIFF file.
+#' f <- 
+#'   gdal_conversion(in_path = in_path, 
+#'     outfolder = outfolder, 
+#'     band_name = band_names[1], 
+#'     just_band_name = F, 
+#'     target_format = ".tif")
+#' }
+gdal_conversion <- function(in_path, outfolder = NULL, band_name = NULL, just_band_name = T, target_format = ".tif") {
+  # check if package has been installed.
+  if (!"gdalUtils" %in% installed.packages()) {
```

Comment:
Thanks for including a check for availability! But `?installed.packages` recommends against this usage:

> It will be slow when thousands of packages are installed, so do not use it to find out if a named package is installed (use [find.package](http://localhost:31789/library/utils/help/find.package) or [system.file](http://localhost:31789/library/utils/help/system.file)) nor to find out if a package is usable (call [requireNamespace](http://localhost:31789/library/utils/help/requireNamespace) or [require](http://localhost:31789/library/utils/help/require) and check the return value) nor to find details of a small number of packages (use [packageDescription](http://localhost:31789/library/utils/help/packageDescription)).

I usually use requireNamespace for this:

```suggestion
  if (!requireNamespace("gdalUtils", quietly = TRUE)) {
```

Reply:
Fixed.